shell/components/form/LabeledSelect.vue

```diff
@@ -178,6 +179,19 @@ export default {
       }
     },
 
+    // If the option's label changed in parent but value did not, the label wont be automatically updated here
+    // Ensure that the label being shown is still present in the options prop and find the new one if not
+    getUpdatedOption(option) {
+      let out;
+      const isOutdated = !this.options.find((opt) => option[this.optionLabel] === opt[this.optionLabel]);
+
+      if (isOutdated && this.options) {
+        out = this.options.find((opt) => isEqual(this.reduce(option), this.reduce(opt)));
+      }
+
+      return out;
+    },
```

Comment:
Since we not reassign it, would be more convenient to use a constant, otherwise it may confuse.

Also if we check `this.options`, we should do it before find out about outdated values.

```ts
      const isOutdated = this.options && !this.options.find((opt) => option[this.optionLabel] === opt[this.optionLabel]);

      return isOutdated
         ? this.options.find((opt) => isEqual(this.reduce(option), this.reduce(opt)))
         : undefined;
```

Reply:
Makes sense, done